### ISSUE TITLE
Sometimes DL License Scanner  crashed on iPhone6

### ISSUE THREAD

**Issue by sirius2013** [PDF417/pdf417-phonegap#23]
Sometimes DL License Scanner  crashed on iPhone6.

Did you have any experience with this issue before?
Thank you

**Comment by DoDoENT** [comment]
None so far. Does it always happen with same DL? If yes, can you provide an image of the barcode? Also, can you provide stack trace of the crash?

**Comment by sirius2013** [comment]
![james_dl](https://cloud.githubusercontent.com/assets/5020713/13953403/d1f6b614-f076-11e5-90bb-bc6625f0621b.jpg)

**Comment by sirius2013** [comment]
We got crash with this DL.

**Comment by sirius2013** [comment]
![tn2_dl](https://cloud.githubusercontent.com/assets/5020713/13954052/0f8e4ab6-f07a-11e5-9464-3e7c05e41358.jpeg)

**Comment by DoDoENT** [comment]
Hi!

We tried scanning both barcodes you've attached with latest PhoneDap SDK on our iPhone 6. There were no crashes.

Are you using latest SDK?

Can you send us the stack trace of the crash?

**Comment by sirius2013** [comment]
do you have example phonegap app to test easily on my side?
Thanks

**Comment by sirius2013** [comment]
And Please check on iPhone5 and 5s also.

**Comment by sirius2013** [comment]
![image](https://cloud.githubusercontent.com/assets/5020713/13954408/9a61c0f4-f07b-11e5-92ac-09f5d743a3c9.png)

**Comment by sirius2013** [comment]
okay, let me try after upgrade sdk.
thanks

**Comment by sirius2013** [comment]
[E] internalGetLicenseToken [AppProtection.cpp:389] 2016-03-23 00:49:14.152 Patient Tracker[960:561231] Invalid key length! Key should have 8 blocks of 8 characters!
2016-03-23 00:49:14.194 Patient Tracker[960:561231] Phone default language unsupported, setting English
2016-03-23 00:49:14.201 Patient Tracker[960:561231] THREAD WARNING: ['Pdf417Scanner'] took '66.673096' ms. Plugin should use a background thread.
[E] -[PPCoordinator checkLicenseKey] [PPCoordinator.mm:576] 2016-03-23 00:49:14.958 Patient Tracker[960:561231] Invalid license key J2J6-ITFJ-VCED-4UJP-3U57-4BEO-DJYM-CIEI. Refusing to load native library!
[E] -[PPCoordinator checkLicenseKey] [PPCoordinator.mm:583] 2016-03-23 00:49:14.959 Patient Tracker[960:561231] App ID is com.knowledgecenter.patient, library version 4.2.0
[E] validateRecognizerSettingsArray [RecognizerSettingsValidator.cpp:55] 2016-03-23 00:49:14.960 Patient Tracker[960:561320] 
Licence token is either null or invalid. You need a valid license key to use the SDK.

FREE demo license keys can be obtained on http://www.microblink.com!

Check that:
1. You generated a demo license key on http://www.microblink.com, or use otherwise obtained license key
2. You use the key for the right platform (don't mix iOS and Android keys)
3. You use the key for the right bundle identifier (License key is linked to exactly one Bundle ID, no wildcards)
2016-03-23 00:49:14.966 Patient Tracker[960:561320] *** Terminating app due to uncaught exception 'Recognizer initialization failed', reason: '
Licence token is either null or invalid. You need a valid license key to use the SDK.

FREE demo license keys can be obtained on http://www.microblink.com!

Check that:
1. You generated a demo license key on http://www.microblink.com, or use otherwise obtained license key
2. You use the key for the right platform (don't mix iOS and Android keys)
3. You use the key for the right bundle identifier (License key is linked to exactly one Bundle ID, no wildcards)

'
*** First throw call stack:
(0x1836c9900 0x182d37f80 0x10028d0c8 0x100731bf0 0x100731bb0 0x10073e6c8 0x1007358a0 0x100731bb0 0x100740e10 0x1007404d8 0x183331470 0x183331020)
libc++abi.dylib: terminating with uncaught exception of type NSException
(lldb)

**Comment by sirius2013** [comment]
and We used the paid license key several months ago.
we need to buy again for this new SDK?

**Comment by DoDoENT** [comment]
Hi, you can generate trial licence key on [MicroBlink Developer Dashboard](https://microblink.com/login).

For updating your paid licence key, please contact sales@microblink.com

**Comment by Cerovec** [comment]
Hi,

did you manage to resolve the issue with the new license key?

**Comment by sirius2013** [comment]
Yes. Scan works, but it works only one time, If I tried to scan again, then scan is not working. 
here is  console log.
hope this is helpful.

Patient Tracker[287:60680] Warning: Attempt to present <CDVCameraPicker: 0x125965e00> on <MainViewController: 0x1256454c0> whose view is not in the window hierarchy!
Patient Tracker[287:60680] Warning: Attempt to present <CameraViewController: 0x126a59570> on <MainViewController: 0x1256454c0> whose view is not in the window hierarchy!
Patient Tracker[287:60680] Warning: Attempt to present <CDVCameraPicker: 0x125969200> on <MainViewController: 0x1256454c0> whose view is not in the window hierarchy!
Patient Tracker[287:60680] Error in Success callbackId: Pdf417Scanner326430938 : ReferenceError: Can't find variable: hex2a
Patient Tracker[287:60680] Snapshotting a view that has not been rendered results in an empty snapshot. Ensure your view has been rendered at least once before snapshotting or snapshot after screen updates.

**Comment by Cerovec** [comment]
Hi @sirius2013,
this issue is opened for a long time.
Is this still an issue for you?
Can we help you in any way?
Jurica.
